dvc/tree/repo.py

```diff
@@ -336,13 +342,17 @@ def walk(
                 onerror(NotADirectoryError(top))
             return
 
+        ignore_subrepos = not self._traverse_subrepos
+        if follow_subrepos is not None:
+            ignore_subrepos = not follow_subrepos
```

Comment:
This is a bit confusing.

Reply:
@skshetry Agreed. I think `_traverse_subrepos` might go away in favor of explicit follow_subrepos + ignore_subrepos.